pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -1060,7 +1060,34 @@ func isKernelPid(pid int) bool {
 	return err != nil && os.IsNotExist(err)
 }
 
+// GetCapacity returns node capacity data for "cpu", "memory", "ephemeral-storage", and "huge-pages*"
+// At present this method is only invoked when introspecting ephemeral storage
 func (cm *containerManagerImpl) GetCapacity() v1.ResourceList {
+	// We store allocatable ephemeral-storage in the capacity property once we Start() the container manager
+	if _, ok := cm.capacity[v1.ResourceEphemeralStorage]; !ok {
+		// If we haven't yet stored the capacity for ephemeral-storage, we can fetch it directly from cAdvisor,
+		// with some retry tolerance in case it hasn't yet started
+		ret := v1.ResourceList{}
+		err := wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
+			if cm.cadvisorInterface != nil {
+				rootfs, err := cm.cadvisorInterface.RootFsInfo()
+				if err == nil {
+					for rName, rQuant := range cm.capacity {
+						ret[rName] = rQuant
+					}
+					ret[v1.ResourceEphemeralStorage] = cadvisor.EphemeralStorageCapacityFromFsInfo(rootfs)[v1.ResourceEphemeralStorage]
+					return true, nil
+				}
+				klog.Errorf("Unable to get rootfs data from cAdvisor interface")
+				return false, nil
```

Comment:
One nit... Usually err is checked for not nil for errors, and the function returns early... IMO the logic should be reversed to take that into account.

It seems like ret should be assigned to cm.capacity, since it contains updated data. Perhaps it was an oversight?

Reply:
inverted the err checking for nil, checking for != nil first now